# **Daffa Asyqar Ahmad Khalisheka**

# **1103200034**

# **Machine Learning Class**

# **Telkom University**

Code di atas digunakan untuk mengimpor library dari Google Colab yang memungkinkan akses ke Google Drive. Selanjutnya, drive.mount('/content/drive') digunakan untuk memasang Google Drive pada direktori /content/drive di lingkungan Google Colab.

In [1]:
# Import library Google Colab untuk mengakses Google Drive
from google.colab import drive

# Mount (pasang) Google Drive pada direktori /content/drive
drive.mount('/content/drive')

Mounted at /content/drive


Code ini adalah sebuah script Python yang digunakan untuk melakukan pemrosesan data dan pembuatan model menggunakan algoritma Neural Network (MLP) untuk klasifikasi. Berikut adalah penjelasan langkah-langkahnya:

1. Mengimpor library yang diperlukan: pandas untuk manipulasi data, train_test_split untuk membagi dataset menjadi data latih dan data uji, StandardScaler untuk penskalaan fitur, LabelEncoder untuk mengubah label kategori menjadi bilangan bulat, accuracy_score untuk mengukur akurasi model, dan MLPClassifier sebagai implementasi Multi-Layer Perceptron untuk klasifikasi.

2. Langkah pertama adalah mengimpor dataset atau data yang akan digunakan untuk pelatihan dan pengujian model. Dataset tersebut kemungkinan besar berupa data yang telah disimpan dalam format yang dapat diolah oleh pandas, seperti CSV atau Excel.

3. Membagi dataset menjadi data latih dan data uji menggunakan fungsi train_test_split. Data latih digunakan untuk melatih model, sedangkan data uji digunakan untuk menguji kinerja model yang telah dilatih.

4. Melakukan penskalaan fitur menggunakan StandardScaler. Hal ini penting untuk memastikan bahwa semua fitur memiliki skala yang seragam, sehingga model dapat belajar dengan baik.

5. Menggunakan LabelEncoder untuk mengubah label kategori menjadi bilangan bulat. Ini diperlukan karena sebagian besar algoritma machine learning memerlukan input label dalam bentuk numerik.

6. Membuat objek MLPClassifier sebagai model Neural Network untuk klasifikasi.

7. Melakukan pelatihan model menggunakan data latih yang telah diproses sebelumnya.

8. Menggunakan model yang telah dilatih untuk melakukan prediksi pada data uji.

9. Mengukur akurasi model dengan membandingkan prediksi model dengan label sebenarnya menggunakan accuracy_score.

Secara keseluruhan, code ini digunakan untuk membangun, melatih, dan menguji model Neural Network untuk tugas klasifikasi pada dataset tertentu.

In [3]:
# Import library yang diperlukan
import pandas as pd
from sklearn.model_selection import train_test_split
from sklearn.preprocessing import StandardScaler, LabelEncoder
from sklearn.metrics import accuracy_score
from sklearn.neural_network import MLPClassifier

Code di atas membaca file CSV yang berisi data dan menyimpannya ke dalam sebuah DataFrame menggunakan library pandas. Path file CSV yang dibaca adalah '/content/drive/MyDrive/Dataset/Weather_Data.csv'. Data tersebut kemudian dimuat ke dalam variabel `data`.

In [4]:
#Membaca file CSV yang berisi data ke dalam sebuah DataFrame
data = pd.read_csv('/content/drive/MyDrive/Dataset/Weather_Data.csv')

Code ini digunakan untuk menampilkan lima baris pertama dari dataset yang disimpan dalam variabel bernama "data." Pemanggilan fungsi "head()" digunakan untuk mengambil sejumlah baris pertama dari dataset tersebut, dan outputnya ditampilkan sebagai hasil eksekusi code.

In [5]:
#Menampilkan isi dari dataset
data.head()

Date  Temp9am  Temp3pm  MinTemp  MaxTemp  Rainfall RainToday  \
0  01-02-13     20.7     20.9     19.5     22.4      15.6       Yes   
1  02-02-13     22.4     24.8     19.5     25.6       6.0       Yes   
2  03-02-13     23.5     23.0     21.6     24.5       6.6       Yes   
3  04-02-13     21.4     20.9     20.2     22.8      18.8       Yes   
4  05-02-13     22.5     25.5     19.7     25.7      77.4       Yes   

   Evaporation  Sunshine WindGustDir  ...  WindDir9am WindDir3pm WindSpeed9am  \
0          6.2       0.0           W  ...           S        SSW           17   
1          3.4       2.7           W  ...           W          E            9   
2          2.4       0.1           W  ...         ESE        ESE           17   
3          2.2       0.0           W  ...         NNE          E           22   
4          4.8       0.0           W  ...         NNE          W           11   

   WindSpeed3pm  Humidity9am  Humidity3pm  Pressure9am  Pressure3pm  Cloud9am  \
0            20           92           84       1017.6       1017.4         8   
1            13           83           73       1017.9       1016.4         7   
2             2           88           86       1016.7       1015.6         7   
3            20           83           90       1014.2       1011.8         8   
4             6           88           74       1008.3       1004.8         8   

   Cloud3pm  
0         8  
1         7  
2         8  
3         8  
4         8  

[5 rows x 21 columns]

Output di atas merupakan tampilan dari lima baris pertama dari dataset. Setiap baris mewakili data yang terkait dengan pengamatan pada tanggal tertentu. Kolom-kolom tersebut mencakup informasi seperti tanggal, suhu pada pukul 9 pagi (Temp9am) dan 3 sore (Temp3pm), suhu minimal (MinTemp) dan maksimal (MaxTemp), jumlah curah hujan (Rainfall), apakah hujan hari ini (RainToday), evaporasi (Evaporation), jumlah jam matahari (Sunshine), arah dan kecepatan angin pada pukul 9 pagi dan 3 sore, kelembaban pada pukul 9 pagi dan 3 sore, tekanan udara pada pukul 9 pagi dan 3 sore, dan informasi awan pada pukul 9 pagi dan 3 sore (Cloud9am dan Cloud3pm).

Code tersebut digunakan untuk menampilkan fitur atau kolom-kolom yang ada dalam dataset. `data.columns` mengembalikan daftar dengan nama-nama kolom yang ada dalam dataset, dan kemudian hasilnya dicetak menggunakan fungsi `print`.

In [6]:
# Tampilkan fitur dalam dataset
print(data.columns)

Index(['Date', 'Temp9am', 'Temp3pm', 'MinTemp', 'MaxTemp', 'Rainfall',
       'RainToday', 'Evaporation', 'Sunshine', 'WindGustDir', 'WindGustSpeed',
       'WindDir9am', 'WindDir3pm', 'WindSpeed9am', 'WindSpeed3pm',
       'Humidity9am', 'Humidity3pm', 'Pressure9am', 'Pressure3pm', 'Cloud9am',
       'Cloud3pm'],
      dtype='object')


Output di atas menunjukkan nama-nama kolom atau fitur dalam dataset. Berikut adalah penjelasan singkat untuk setiap kolom:

1. **Date**: Menunjukkan tanggal pengamatan.
2. **Temp9am**: Suhu pada pukul 9 pagi.
3. **Temp3pm**: Suhu pada pukul 3 sore.
4. **MinTemp**: Suhu minimum pada hari itu.
5. **MaxTemp**: Suhu maksimum pada hari itu.
6. **Rainfall**: Jumlah curah hujan dalam milimeter.
7. **RainToday**: Menunjukkan apakah hujan terjadi hari itu (Yes/No).
8. **Evaporation**: Penguapan dalam milimeter.
9. **Sunshine**: Jumlah jam sinar matahari.
10. **WindGustDir**: Arah angin pada kejadian angin kencang.
11. **WindGustSpeed**: Kecepatan angin saat terjadi angin kencang.
12. **WindDir9am**: Arah angin pada pukul 9 pagi.
13. **WindDir3pm**: Arah angin pada pukul 3 sore.
14. **WindSpeed9am**: Kecepatan angin pada pukul 9 pagi.
15. **WindSpeed3pm**: Kecepatan angin pada pukul 3 sore.
16. **Humidity9am**: Kelembapan pada pukul 9 pagi.
17. **Humidity3pm**: Kelembapan pada pukul 3 sore.
18. **Pressure9am**: Tekanan udara pada pukul 9 pagi.
19. **Pressure3pm**: Tekanan udara pada pukul 3 sore.
20. **Cloud9am**: Jumlah awan pada pukul 9 pagi.
21. **Cloud3pm**: Jumlah awan pada pukul 3 sore.

Jenis data (dtype) untuk setiap kolom tidak ditampilkan dalam output tersebut. Dalam analisis data yang lebih mendalam, informasi tersebut dapat penting untuk memahami dan memanipulasi data dengan benar.

Kode tersebut mengakses metode `info()` pada objek yang disebut 'data'. Metode ini digunakan untuk memberikan informasi yang lengkap tentang struktur dan karakteristik objek data tersebut. Metode `info()` biasanya digunakan pada objek data seperti DataFrame dalam library pandas pada bahasa pemrograman Python. Kode ini akan memberikan output yang mencakup informasi seperti jumlah kolom, nama kolom, tipe data setiap kolom, jumlah nilai non-null, dan penggunaan memori oleh objek data tersebut. Informasi ini sangat bermanfaat untuk memahami dan menganalisis data yang sedang diproses.

In [7]:
# Ini adalah kode untuk menampilkan informasi tentang objek 'data'
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3271 entries, 0 to 3270
Data columns (total 21 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Date           3271 non-null   object 
 1   Temp9am        3271 non-null   float64
 2   Temp3pm        3271 non-null   float64
 3   MinTemp        3271 non-null   float64
 4   MaxTemp        3271 non-null   float64
 5   Rainfall       3271 non-null   float64
 6   RainToday      3271 non-null   object 
 7   Evaporation    3271 non-null   float64
 8   Sunshine       3271 non-null   float64
 9   WindGustDir    3271 non-null   object 
 10  WindGustSpeed  3271 non-null   int64  
 11  WindDir9am     3271 non-null   object 
 12  WindDir3pm     3271 non-null   object 
 13  WindSpeed9am   3271 non-null   int64  
 14  WindSpeed3pm   3271 non-null   int64  
 15  Humidity9am    3271 non-null   int64  
 16  Humidity3pm    3271 non-null   int64  
 17  Pressure9am    3271 non-null   float64
 18  Pressure

Output di atas memberikan informasi rinci tentang sebuah DataFrame dalam library pandas pada bahasa pemrograman Python. Berikut adalah penjelasan dari setiap bagian output tersebut:

1. **Class 'pandas.core.frame.DataFrame':**
   - Menunjukkan bahwa objek ini adalah suatu DataFrame dalam library pandas.

2. **RangeIndex: 3271 entries, 0 to 3270:**
   - Menunjukkan bahwa DataFrame ini memiliki RangeIndex dengan total 3271 baris, dimulai dari indeks 0 hingga 3270.

3. **Data columns (total 21 columns):**
   - Memberikan informasi bahwa DataFrame ini memiliki 21 kolom.

4. **Column Information:**
   - Setiap kolom dijelaskan dengan detail sebagai berikut:
     - **Column:** Nama kolom.
     - **Non-Null Count:** Jumlah nilai non-null dalam kolom.
     - **Dtype:** Tipe data kolom.

5. **dtypes:**
   - Memberikan informasi tentang tipe data dari keseluruhan DataFrame, termasuk jumlah kolom untuk setiap tipe data.

6. **memory usage: 536.8+ KB:**
   - Menunjukkan penggunaan memori oleh DataFrame, dalam hal ini sekitar 536.8 kilobyte.

Ringkasnya, output ini memberikan gambaran lengkap tentang struktur DataFrame, termasuk jumlah baris dan kolom, informasi tentang setiap kolom seperti nama, jumlah nilai non-null, dan tipe datanya, serta penggunaan memori oleh DataFrame tersebut. Output ini berguna untuk memahami dan menganalisis data yang dimuat dalam DataFrame.

```python
# Menghapus baris yang memiliki nilai null
data = data.dropna()
```

Code tersebut berfungsi untuk menghapus baris atau entri dalam dataset yang mengandung nilai null atau NaN (Not a Number). `data` merupakan nama variabel yang menampung dataset, dan fungsi `dropna()` digunakan untuk menghapus baris yang mengandung setidaknya satu nilai null. Operasi ini bertujuan untuk membersihkan dataset dari data yang tidak lengkap atau tidak tersedia sehingga analisis atau pemrosesan data selanjutnya dapat dilakukan dengan data yang lebih lengkap dan akurat.

In [8]:
# Menghapus baris yang memiliki nilai null
data= data.dropna()

Code tersebut digunakan untuk memisahkan fitur (X) dan target (y) dari suatu dataset. Variabel `X` akan berisi data yang tidak termasuk kolom yang disebutkan dalam fungsi `drop` (dalam hal ini, kolom 'Rainfall'), sementara variabel `y` akan berisi data dari kolom yang disebutkan (dalam hal ini, 'RainToday'). Hal ini berguna dalam persiapan data untuk pemodelan, di mana fitur-fitur akan digunakan untuk memprediksi target. Code tersebut mengasumsikan penggunaan Google Colab, suatu lingkungan komputasi berbasis cloud untuk pemrograman Python.

In [13]:
# Pisahkan fitur (X) dan target (y)
X = data.drop('Rainfall', axis=1)  # Ganti 'RainTomorrow' dengan nama kolom target yang sesuai
y = data['RainToday']


Potongan kode yang diberikan menggunakan kelas `LabelEncoder` dari scikit-learn untuk melakukan encoding pada variabel kategori dalam DataFrame `X`. Secara khusus, variabel 'RainToday', 'WindGustDir', 'WindDir9am', dan 'WindDir3pm' di-encode. Metode `fit_transform` diterapkan pada setiap kolom untuk mengonfigurasi encoder dan mengubah nilai kategori menjadi label numerik. Proses ini umumnya digunakan dalam pembelajaran mesin untuk mengonversi data kategori ke dalam format yang dapat digunakan oleh algoritma pembelajaran mesin, karena banyak algoritma bekerja dengan input numerik. Nilai-nilai yang telah diencode kemudian diberikan kembali ke kolom-kolom mereka yang sesuai dalam DataFrame. Kode ini berguna untuk mempersiapkan data kategori agar dapat digunakan oleh model pembelajaran mesin.

In [14]:
# Lakukan encoding untuk variabel kategori
le = LabelEncoder()
X['RainToday'] = le.fit_transform(X['RainToday'])
X['WindGustDir'] = le.fit_transform(X['WindGustDir'])
X['WindDir9am'] = le.fit_transform(X['WindDir9am'])
X['WindDir3pm'] = le.fit_transform(X['WindDir3pm'])

Kode tersebut menggunakan fungsi `train_test_split` dari pustaka scikit-learn untuk membagi suatu dataset menjadi dua bagian: data latih dan data uji. Dataset ini terdiri dari fitur (X) dan label (y). Pembagian dilakukan dengan ukuran data uji sebesar 20%, yang berarti 80% data akan digunakan untuk melatih model, dan 20% untuk menguji model. Parameter `random_state` digunakan untuk memastikan reproduktibilitas dengan menetapkan seed untuk generator angka acak. Hasil pemisahan disimpan dalam variabel `X_train`, `X_test`, `y_train`, dan `y_test`. Ini merupakan praktik umum dalam pembelajaran mesin untuk mengevaluasi kinerja model pada data yang tidak terlihat selama pelatihan. Kode tersebut diasumsikan dijalankan di lingkungan Google Colab.

In [15]:
# Bagi dataset menjadi data latih dan data uji
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)


```python
print(X_train.dtypes)
```

Code ini digunakan untuk mencetak jenis data (tipe data) dari setiap kolom dalam DataFrame `X_train`. Ini membantu dalam memahami tipe data yang digunakan dalam setiap kolom, seperti apakah itu integer, float, string, atau tipe data lainnya. Output dari code ini akan memberikan informasi tentang jenis data yang ada dalam setiap kolom dari DataFrame tersebut.

In [23]:
print(X_train.dtypes)


Temp9am          float64
Temp3pm          float64
MinTemp          float64
MaxTemp          float64
RainToday          int64
Evaporation      float64
Sunshine         float64
WindGustDir       object
WindGustSpeed      int64
WindDir9am        object
WindDir3pm        object
WindSpeed9am       int64
WindSpeed3pm       int64
Humidity9am        int64
Humidity3pm        int64
Pressure9am      float64
Pressure3pm      float64
Cloud9am           int64
Cloud3pm           int64
dtype: object


Output di atas adalah hasil dari pemanggilan `print(X_train.dtypes)` pada suatu DataFrame, yang menunjukkan jenis data (tipe data) dari setiap kolom dalam DataFrame tersebut. Berikut adalah penjelasan singkat untuk setiap kolom:

1. `Temp9am`, `Temp3pm`, `MinTemp`, `MaxTemp`, `Evaporation`, `Sunshine`, `Pressure9am`, `Pressure3pm`: Kolom-kolom ini memiliki tipe data float64, yang menunjukkan bahwa nilai-nilai dalam kolom tersebut berupa bilangan pecahan.

2. `RainToday`, `WindGustSpeed`, `WindSpeed9am`, `WindSpeed3pm`, `Humidity9am`, `Humidity3pm`, `Cloud9am`, `Cloud3pm`: Kolom-kolom ini memiliki tipe data int64, yang menunjukkan bahwa nilai-nilai dalam kolom tersebut berupa bilangan bulat.

3. `WindGustDir`, `WindDir9am`, `WindDir3pm`: Kolom-kolom ini memiliki tipe data object, yang mungkin menunjukkan bahwa nilai-nilai dalam kolom tersebut adalah string atau data kategorikal.

Dengan mengetahui jenis data setiap kolom, kita dapat memahami format data apa yang diharapkan dan melakukan pengolahan data yang sesuai. Misalnya, kolom-kolom dengan tipe data object mungkin perlu diubah menjadi tipe data kategorikal atau diolah lebih lanjut untuk analisis data.

Kode tersebut menggunakan pustaka pandas untuk membuat DataFrames dari array NumPy X_train dan X_test dengan nama kolom yang dibuat secara dinamis. Kolom kategorikal diidentifikasi dan di-encode dengan one-hot encoding. Set uji kemudian dimodifikasi agar memiliki kolom yang sama dengan set pelatihan setelah one-hot encoding. Terakhir, fitur-fitur dinormalisasi menggunakan StandardScaler dari scikit-learn.

In [29]:
import pandas as pd
from sklearn.preprocessing import StandardScaler

# Assuming you have X_train and X_test as NumPy arrays
column_names = [f"Feature_{i}" for i in range(X_train.shape[1])]

# Convert X_train and X_test to DataFrames with dynamically generated column names
X_train_df = pd.DataFrame(X_train, columns=column_names)
X_test_df = pd.DataFrame(X_test, columns=column_names)  # Ensure the same columns as X_train

# Identify and one-hot encode categorical columns
categorical_cols = X_train_df.select_dtypes(include=['object']).columns
X_train_df = pd.get_dummies(X_train_df, columns=categorical_cols, drop_first=True)
X_test_df = pd.get_dummies(X_test_df, columns=categorical_cols, drop_first=True)

# Ensure that the test set has the same columns as the training set after one-hot encoding
missing_cols = set(X_train_df.columns) - set(X_test_df.columns)
for col in missing_cols:
    X_test_df[col] = 0

# Standardize the features
scaler = StandardScaler()
X_train_scaled = scaler.fit_transform(X_train_df)
X_test_scaled = scaler.transform(X_test_df)



Kode yang diberikan adalah sebuah skrip Python yang menggunakan pustaka scikit-learn untuk pra-pemrosesan data dalam suatu pipa garis pembelajaran mesin. Berikut adalah penjelasan rinci dari setiap bagian:

1. Mengimpor pustaka yang diperlukan:
   - `LabelEncoder` dan `StandardScaler` dari `sklearn.preprocessing`
   - `train_test_split` dari `sklearn.model_selection`

2. Memilih kolom numerik untuk distandardisasi:
   - Mengidentifikasi kolom numerik dalam data masukan `X` menggunakan `select_dtypes`.
   - Membuat DataFrame baru `X_numeric` yang hanya berisi kolom numerik.

3. Membagi dataset menjadi set pelatihan dan pengujian:
   - Menggunakan `train_test_split` untuk membagi data menjadi set pelatihan dan pengujian (`X_train`, `X_test`, `y_train`, `y_test`).
   - Rasio pembagian adalah 80% untuk pelatihan dan 20% untuk pengujian, dan `random_state` diatur agar hasil dapat direproduksi.

4. Mengkodekan nilai kategorikal dalam kolom 'RainToday':
   - Menggunakan `LabelEncoder` (`le`) untuk mentransformasi nilai kategorikal dalam kolom 'RainToday' dari set pelatihan dan pengujian menjadi nilai numerik.
   - Transformasi diterapkan pada set pelatihan dan pengujian secara terpisah.

5. Menstandardisasi fitur numerik:
   - Menggunakan `StandardScaler` (`scaler`) untuk menstandardisasi fitur numerik dari set pelatihan dan pengujian.
   - `fit_transform` digunakan pada set pelatihan untuk menghitung rata-rata dan deviasi standar untuk standardisasi dan menerapkan transformasi.
   - `transform` digunakan pada set pengujian untuk menerapkan transformasi yang sama menggunakan parameter yang dipelajari dari set pelatihan.

Catatan: Kode mengasumsikan bahwa data masukan (`X`) memiliki kolom bernama 'RainToday' yang berisi nilai kategorikal. Pastikan data yang diperlukan sudah dimuat sebelum menjalankan kode ini. Selain itu, tampaknya ada masalah dengan mengubah `X_train` dan `X_test` langsung saat mengkodekan nilai kategorikal. Lebih aman untuk membuat salinan DataFrame sebelum melakukan modifikasi tersebut untuk menghindari efek samping potensial.

In [31]:
from sklearn.preprocessing import LabelEncoder, StandardScaler
from sklearn.model_selection import train_test_split

# Exclude non-numeric columns from standardization
numeric_cols = X.select_dtypes(include='number').columns
X_numeric = X[numeric_cols]

# Split the dataset into training and testing sets
X_train, X_test, y_train, y_test = train_test_split(X_numeric, y, test_size=0.2, random_state=42)

# Encode categorical values in 'RainToday' column
le = LabelEncoder()
X_train['RainToday'] = le.fit_transform(X_train['RainToday'])
X_test['RainToday'] = le.transform(X_test['RainToday'])

# Standardize the numeric features
scaler = StandardScaler()
X_train_scaled = scaler.fit_transform(X_train)
X_test_scaled = scaler.transform(X_test)


Kode yang diberikan menggunakan pustaka scikit-learn untuk melakukan pra-pemrosesan data dan pembagian data latih dan uji untuk model pembelajaran mesin. Kode tersebut mengasumsikan adanya DataFrame dengan nama 'data' yang berisi fitur dan variabel target ('RainToday'). Tujuannya adalah untuk mempersiapkan data untuk melatih model pembelajaran mesin dengan menghapus kolom non-numerik, membagi data menjadi set latih dan uji, serta menstandarisasi fitur numerik menggunakan StandardScaler dari scikit-learn. Parameter random_state diatur untuk reproduksibilitas. Harap dicatat bahwa kode ini dimaksudkan untuk dieksekusi di lingkungan seperti Jupyter notebook atau Google Colab.

In [32]:
from sklearn.preprocessing import StandardScaler
from sklearn.model_selection import train_test_split

# Assuming you have a DataFrame 'data' with features and target
# Exclude non-numeric columns ('Date') from features
X = data.drop(['Rainfall', 'Date'], axis=1)
y = data['RainToday']

# Split the dataset into training and testing sets
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)

# Identify numeric columns and standardize them
numeric_cols = X_train.select_dtypes(include='number').columns
scaler = StandardScaler()
X_train[numeric_cols] = scaler.fit_transform(X_train[numeric_cols])
X_test[numeric_cols] = scaler.transform(X_test[numeric_cols])


Kode tersebut mengimpor pustaka dan modul yang diperlukan dari scikit-learn dan pandas. Kode mengasumsikan adanya DataFrame yang dinamai 'data' yang berisi fitur dan variabel target. Kolom non-numerik ('Date') dikecualikan dari fitur. Kolom kategorikal diidentifikasi, dan one-hot encoding diterapkan pada mereka. Dataset kemudian dibagi menjadi set pelatihan dan pengujian. Terakhir, model jaringan saraf (MLPClassifier) diinisialisasi dan dilatih menggunakan data pelatihan. Model memiliki satu lapisan tersembunyi dengan 100 neuron, maksimum 1000 iterasi, dan seed acak untuk reproduktibilitas.

In [35]:
from sklearn.preprocessing import OneHotEncoder
from sklearn.neural_network import MLPClassifier
from sklearn.model_selection import train_test_split
import pandas as pd

# Assuming you have a DataFrame 'data' with features and target
# Exclude non-numeric columns ('Date') from features
X = data.drop(['Rainfall', 'Date'], axis=1)
y = data['RainToday']

# Identify and one-hot encode categorical columns
categorical_cols = X.select_dtypes(include=['object']).columns
X = pd.get_dummies(X, columns=categorical_cols, drop_first=True)

# Split the dataset into training and testing sets
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)

# Inisialisasi dan latih model Neural Network
model = MLPClassifier(hidden_layer_sizes=(100,), max_iter=1000, random_state=42)
model.fit(X_train, y_train)



MLPClassifier(max_iter=1000, random_state=42)

`MLPClassifier` adalah kelas dalam pustaka scikit-learn yang mewakili implementasi dari jaringan saraf dengan pembelajaran yang mendalam atau lebih dikenal sebagai Multi-Layer Perceptron (MLP). Di sini, ada dua parameter yang diberikan saat menginisialisasi objek MLPClassifier:

1. `max_iter=1000`: Ini adalah parameter yang menentukan jumlah maksimum iterasi (epoch) yang akan dilakukan oleh optimizer ketika melatih model. Dalam konteks ini, proses pelatihan akan dihentikan setelah 1000 iterasi atau epoch. Jumlah iterasi ini dapat mempengaruhi kualitas dan konvergensi model.

2. `random_state=42`: Ini adalah parameter yang menetapkan seed untuk generator angka acak. Menetapkan seed acak memastikan bahwa proses pelatihan jaringan saraf dapat direproduksi dengan hasil yang sama setiap kali kode dijalankan. Dengan kata lain, jika Anda menjalankan kode ini beberapa kali dengan seed acak yang sama, Anda akan mendapatkan hasil yang konsisten. Ini berguna untuk tujuan reproduktibilitas dan pemecahan masalah.

Jadi, output di atas menciptakan objek MLPClassifier dengan konfigurasi tersebut untuk digunakan dalam pelatihan model jaringan saraf.

```python
# Lakukan prediksi pada data uji
y_pred = model.predict(X_test)
```

Code tersebut melakukan prediksi menggunakan model yang telah di-train pada data uji (`X_test`). Hasil prediksi disimpan dalam variabel `y_pred`.

In [36]:
# Lakukan prediksi pada data uji
y_pred = model.predict(X_test)

```python
# Menghitung akurasi model dengan menggunakan fungsi accuracy_score dari library scikit-learn
accuracy = accuracy_score(y_test, y_pred)

# Menampilkan hasil akurasi model dalam bentuk persentase dengan dua angka desimal di belakang koma
print(f'Akurasi Model: {accuracy * 100:.2f}%')
```

Code tersebut menghitung akurasi dari model machine learning dengan menggunakan fungsi `accuracy_score` dari library scikit-learn. Akurasi dihitung dengan membandingkan nilai yang diprediksi oleh model (`y_pred`) dengan nilai sebenarnya (`y_test`). Hasil akurasi kemudian ditampilkan dalam bentuk persentase dengan dua angka desimal di belakang koma menggunakan fungsi `print`.

In [37]:
# Evaluasi model
accuracy = accuracy_score(y_test, y_pred)
print(f'Akurasi Model: {accuracy * 100:.2f}%')

Akurasi Model: 100.00%


Output "Akurasi Model: 100.00%" menunjukkan bahwa model machine learning yang telah dibuat memberikan prediksi yang sempurna atau tepat sesuai dengan nilai sebenarnya pada data uji (`y_test`). Dengan kata lain, setiap prediksi yang dihasilkan oleh model sama dengan nilai sebenarnya, sehingga akurasi model mencapai 100%.

Akurasi merupakan metrik evaluasi yang mengukur sejauh mana model dapat membuat prediksi yang benar. Dalam konteks ini, akurasi sebesar 100% menunjukkan bahwa tidak ada satu pun sampel di data uji yang memiliki prediksi yang salah oleh model. Meskipun hasil ini terlihat sangat baik, perlu diingat bahwa mencapai akurasi 100% juga dapat menimbulkan kecurigaan, terutama jika tidak ada langkah-langkah pengujian yang memadai dan jika model belum diuji dengan dataset yang lebih luas.

Penting untuk memeriksa performa model secara menyeluruh menggunakan berbagai metrik evaluasi dan menguji model pada dataset yang berbeda untuk memastikan kehandalan dan generalitasnya. Akurasi sendiri mungkin tidak mencukupi dalam beberapa kasus, terutama jika data tidak seimbang atau jika terdapat kelas minoritas yang penting untuk diidentifikasi dengan akurat.